forc/src/ops/forc_build.rs

```diff
@@ -42,7 +42,10 @@ pub fn build(command: BuildCommand) -> Result<pkg::Compiled> {
     let lock_path = lock_path(manifest.dir());
 
     // Load the build plan from the lock file.
-    let plan_result = pkg::BuildPlan::from_lock_file(&lock_path, SWAY_GIT_TAG);
+    // This BuildPlan would be unsafe (true as passed as unsafe_plan) meaning
+    // there might be some pinned dependency whose parent does not include itself,
+    // i.e: a removed dependency. For more information, please see https://github.com/FuelLabs/sway/issues/1778
+    let plan_result = pkg::BuildPlan::from_lock_file(&lock_path, SWAY_GIT_TAG, true);
```

Comment:
I can see where you're coming from with this change w.r.t. attempting to fix #1778, but I don't think we should enable the creation of a potentially unsafe build plan. We should always uphold all invariants for each of our types to avoid ever ending up in a position where we're unsure if they're valid or not, otherwise our types lose meaning.

Would you mind removing the `unsafe_plan` changes from this PR and we can discuss the solution in #1778?

Reply:
That makes perfect sense, but without fixing #1778, `apply_new_manifest` will not be able to handle removed dependencies as we wanted (once there is a removal, BuildPlan will get generated from scratch). Should we discuss and solve it in #1778 and merge it here or once this PR is ready should we merge this one and merge solution for the #1778 later?